lib/sw-precache.js

```diff
@@ -47,9 +51,13 @@ function getFileAndSizeAndHashForFile(file) {
   return null;
 }
 
-function getFilesAndSizesAndHashesForGlobPattern(globPattern) {
+function getFilesAndSizesAndHashesForGlobPattern(globPattern, excludeFilePath) {
   return glob.sync(globPattern.replace(path.sep, '/')).map(function(file) {
-    return getFileAndSizeAndHashForFile(file);
+    // Return null if we want to exclude this file, and it will be excluded in
+    // the subsequent filter().
+    return excludeFilePath === absolutePath(file) ?
+      null :
```

Comment:
You save an unneeded `fs.statSync()` by filtering it out early and avoiding the call to `getFileAndSizeAndHashForFile()`. That was the main reason.